File: backend/EasyCourse.Core/Entities/Comment.cs
using System.Text.Json.Serialization;

namespace EasyCourse.Core.Entities;

public class Comment
{
    public int Id { get; set; }
    public string Text { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public Guid UserId { get; set; }
    [JsonIgnore]
    public User User { get; set; } = null!;
    public string EntityId { get; set; } = string.Empty;
    public string EntityType { get; set; } = string.Empty;
}

File: frontend/src/types/comment.ts
export interface Comment {
  id?: string;
  userId: string;
  createdAt: Date;
  updatedAt?: Date;
  text?: string;
  entityId: string;
  entityType: string;
  username: string,
  userImagePath?: string;
}

export interface CommentRequest {
  text: string;
  entityId: string;
  entityType: string;
  userId: string;
}
